internal/events/event.go

```diff
@@ -194,6 +208,12 @@ type ClusterEvent interface {
 
 type HostEvent interface {
 	BaseEvent
-	GetClusterId() strfmt.UUID
+	GetClusterId() *strfmt.UUID
 	GetHostId() strfmt.UUID
+	GetInfraEnvId() strfmt.UUID
+}
+
+type InfraEnvEvent interface {
+	BaseEvent
+	GetInfraEnvId() strfmt.UUID
```

Comment:
There is an option that we'd might have cluster-ID for the InfraEnv.
In that case, we'll have to add `GetClusterId() *strfmt.UUID` here as well

Reply:
done